src/components/Form.tsx

```diff
@@ -16,41 +18,54 @@ export const Form: React.FunctionComponent<IFormtronControl> = ({
   fieldComponents,
   path,
   disabled = false,
+  layout,
 }) => {
   const { variant } = useDiagnostics(path);
   const keys = Object.keys(schema.fields);
-  const guts = keys.map((name, index) => {
-    const formId = `${name}-${index}`;
-    const propSchema = schema.fields[name];
-    if (propSchema.show) {
-      const show = evaluate(propSchema.show, value, name, true);
-      if (!show) return null;
-    }
-    if (propSchema.evalOptions) {
-      propSchema.options = evaluate(propSchema.evalOptions, value, name, []);
-    }
-    const Widget = fieldComponents[propSchema.type];
-    if (Widget === undefined) {
-      throw new Error(`No appropriate widget could be found for type "${propSchema.type}"`);
-    }
-    const el = (
-      <div key={formId}>
-        <Widget
-          id={formId}
-          value={value[name]}
-          schema={propSchema}
-          path={replaceWildcards(name, path)}
-          onChange={(val: any) => {
-            const v = { ...value, [name]: val };
-            onChange(v);
-          }}
-          fieldComponents={fieldComponents}
-          disabled={disabled}
-        />
-      </div>
-    );
-    return el;
-  });
+  const grid = layout && schema.layouts && schema.layouts[layout];
+  const guts = (
+    <div
+      style={{
+        display: 'grid',
+        gridTemplateAreas: grid ? toGridTemplateAreas(grid) : undefined,
+        gridColumnGap: '10px',
+        gridRowGap: '10px',
+      }}
+    >
+      {keys.map((name, index) => {
+        const formId = `${name}-${index}`;
+        const propSchema = schema.fields[name];
+        if (propSchema.show) {
+          const show = evaluate(propSchema.show, value, name, true);
```

Comment:
deconstruct propSchema for more readability

`const { show, evalOptions, area, .... } = propSchema`

Reply:
eh... maybe next time. we need to actually set `propSchema.options` and we already have a variable called `show` ...